.circleci/config.yml

```diff
@@ -114,6 +117,27 @@ jobs:
           upload_name: Coverage Report for iOS Tests
           xtra_args: -c -v --xc --xp iosresults.xcresult
   
+  test-ios-integration:
+    macos:
+      xcode: 13.4.0 # Specify the Xcode version to use
+
+    steps:
+      - checkout
+
+      - install_dependencies
+
+      - prestart_ios_simulator
+
+      - run:
+          name: Run iOS integration tests
+          command: |
+            GITHUB_PR_JSON=`curl -s https://api.github.com/repos/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}/pulls/${CIRCLE_PR_NUMBER}`;
+            CIRCLE_PR_BASE_BRANCH=`echo "$GITHUB_PR_JSON" | jq -r '.base.ref'`;
+            echo $CIRCLE_PR_BASE_BRANCH;
+            if [ "$CIRCLE_PR_BASE_BRANCH" == "main" ] || [ "$CIRCLE_PR_BASE_BRANCH" == "staging" ]; then
```

Comment:
can we use a consistent way of retrieving the branch for the steps that need it? would circleci_branch suffice if we check that the triggering branch is prefixed with "dev" or equal to "staging"?

Reply:
For simplicity can we work with this circleci branch retrieval limitation for now and consider this option later if we are to use github workflows instead? Also, you might need to sync the branch base with dev which now checks for both staging and main.

Reply:
Let's not complicate the workflow with additional APIs, let's keep it simple and let run on merge to stage. 

Just a thought: Is it possible to configure a github action to trigger on PR creation to stage or main and run integration tests?